x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/TestCaseSupplier.java

```diff
@@ -138,47 +138,60 @@ public static List<TestCaseSupplier> forBinaryCastingToDouble(
         String name,
         String lhsName,
         String rhsName,
-        DoubleBinaryOperator expected
+        DoubleBinaryOperator expected,
+        Double lhsMin,
+        Double lhsMax,
+        Double rhsMin,
+        Double rhsMax,
+        List<String> warnings
     ) {
         List<TestCaseSupplier> suppliers = new ArrayList<>();
-        for (DataType lhsType : EsqlDataTypes.types()) {
-            if (lhsType.isNumeric() == false || EsqlDataTypes.isRepresentable(lhsType) == false) {
-                continue;
-            }
-            for (Map.Entry<String, Supplier<Object>> lhsSupplier : RANDOM_VALUE_SUPPLIERS.get(lhsType)) {
-                for (DataType rhsType : EsqlDataTypes.types()) {
-                    if (rhsType.isNumeric() == false || EsqlDataTypes.isRepresentable(rhsType) == false) {
-                        continue;
+        List<TypedDataSupplier> lhsSuppliers = new ArrayList<>();
+        List<TypedDataSupplier> rhsSuppliers = new ArrayList<>();
+
+        lhsSuppliers.addAll(intCases(lhsMin.intValue(), lhsMax.intValue()));
+        lhsSuppliers.addAll(longCases(lhsMin.longValue(), lhsMax.longValue()));
+        lhsSuppliers.addAll(ulongCases(BigInteger.valueOf((long) Math.ceil(lhsMin)), BigInteger.valueOf((long) Math.floor(lhsMax))));
+        lhsSuppliers.addAll(doubleCases(lhsMin, lhsMax));
+
+        rhsSuppliers.addAll(intCases(rhsMin.intValue(), rhsMax.intValue()));
+        rhsSuppliers.addAll(longCases(rhsMin.longValue(), rhsMax.longValue()));
+        rhsSuppliers.addAll(ulongCases(BigInteger.valueOf((long) Math.ceil(rhsMin)), BigInteger.valueOf((long) Math.floor(rhsMax))));
+        rhsSuppliers.addAll(doubleCases(rhsMin, rhsMax));
```

Comment:
praise: I like the more explicit approach of listing all the test case suppliers, rather than filtering based on some predicates, since this is easier for me to read and quickly understand what is going to be tested.